build.sbt

```diff
@@ -99,6 +99,66 @@ genBuildInfo := {
   FileUtils.writeStringToFile(infoFile, buildInfo, "utf-8")
 }
 
+// scalastyle:off line.size.limit
+val genSleetConfig = TaskKey[Unit]("genSleetConfig",
+  "generate sleet.json file for sleet configuration so we can push nuget package to the blob")
+genSleetConfig := {
+  val fileContent =
+    s"""{
+       |  "username": "",
+       |  "useremail": "",
+       |  "sources": [
+       |    {
+       |      "name": "SynapseMLNuget",
+       |      "type": "azure",
+       |      "container": "synapsemlnuget",
+       |      "path": "https://mmlspark.blob.core.windows.net/synapsemlnuget",
+       |      "connectionString": "DefaultEndpointsProtocol=https;AccountName=mmlspark;AccountKey=${Secrets.storageKey};EndpointSuffix=core.windows.net"
+       |    }
+       |  ]
+       |}""".stripMargin
+  val sleetJsonFile = join(rootGenDir.value, "sleet.json")
+  if (sleetJsonFile.exists()) FileUtils.forceDelete(sleetJsonFile)
+  FileUtils.writeStringToFile(sleetJsonFile, fileContent, "utf-8")
+}
+// scalastyle:on line.size.limit
+
+val publishDotnetTestBase = TaskKey[Unit]("publishDotnetTestBase",
+  "generate dotnet test helper file with current library version and publish E2E test base")
+publishDotnetTestBase := {
+  genSleetConfig.value
+  val fileContent =
+    s"""// Licensed to the .NET Foundation under one or more agreements.
+       |// The .NET Foundation licenses this file to you under the MIT license.
+       |// See the LICENSE file in the project root for more information.
+       |
+       |namespace SynapseMLtest.Utils
+       |{
+       |    public class Helper
+       |    {
+       |        public static string GetSynapseMLPackage()
+       |        {
+       |            return "com.microsoft.azure:synapseml_2.12:${version.value}";
+       |        }
+       |    }
+       |
+       |}
+       |""".stripMargin
+  val dotnetTestBaseDir = join(baseDirectory.value, "core", "src", "main", "dotnet", "test")
+  val dotnetHelperFile = join(dotnetTestBaseDir, "SynapseMLVersion.cs")
+  if (dotnetHelperFile.exists()) FileUtils.forceDelete(dotnetHelperFile)
+  FileUtils.writeStringToFile(dotnetHelperFile, fileContent, "utf-8")
+  runCmd(
+    Seq("dotnet", "pack", "--output", join(dotnetTestBaseDir, "target").getAbsolutePath),
+    dotnetTestBaseDir
+  )
+  val packagePath = join(dotnetTestBaseDir, "target", s"SynapseML.DotnetE2ETest.0.9.1.nupkg").getAbsolutePath
```

Comment:
I fix this version on purpose, because the only change to the dotnetTestBase is the version of synapseml jar that it depends on. This is only used for our internal tests, so I'm open to either, releasing different snapshot versions is also fine.